Serialization.md

```diff
@@ -0,0 +1,184 @@
+
+## Project (De)Serialization
+
+Let's talk first for a PK-Sim project. In [ProjectMetaDataToProjectMapper.cs](https://github.com/Open-Systems-Pharmacology/PK-Sim/blob/develop/src/PKSim.Infrastructure/Serialization/ORM/Mappers/ProjectMetaDataToProjectMapper.cs) of the PK-Sim solution the order of (de)serialization is defined. When starting the application e.g. we get the project structure from the database and then we further load the contents from the project file. In MoBi correspondingly the project (de)serialization is also defined in [ProjectMetaDataToProjectMapper.cs](https://github.com/Open-Systems-Pharmacology/MoBi/blob/develop/src/MoBi.Core/Serialization/ORM/Mappers/ProjectMetaDataToProjectMapper.cs) of the MoBi solution. The main difference between the two is that the entities in PK-Sim are lazy loaded, whereas in MoBi we load everything on startup.
+
+## Writing a serializer for a new class
+
+When creating a new class in OSPSuite of an object that will then need to be saved to the project file, a new serializer will also have to be written for that class. Let's call our new class `NewClass`. If the class gets created in OSPSuite.Core and is not implementing an interface that already has an abstract serializer, the convention would be to write a serializer called `NewClassSerializer : OSPSuiteXmlSerializer<NewClass>`. 
+
+You will then have to write a an override for the `PerformMapping()` function, that serializes the properties of the class:
+
+For example:
+
+```
+public class NewClass
+{
+   public string Name { get; set; }
+
+}
+```
+
+```
+public class NewClassSerializer : OSPSuiteXmlSerializer<NewClass>
+{
+   public override void PerformMapping()
+   {
+      Map(x => x.Name);
+   }
+}
+```
+
+In case your new class implements an interface that already has an abstract serializer, it is that one that you will need to extend. For example if you are writing a new Building Block class (that would be implementing the interface IBuildingBlock), the serializer signature would be  
+
+```
+public class NewClassSerializer : BuildingBlockXmlSerializer<NewClass>
+```
+
+
+Now let's talk a bit about the mapping a classes properties in the `PerformMapping()` override. The most frequent use cases would be:
+
+# Map(...)
```

Comment:
also, I have not included the full signature of the function in the titles because I think that would make the documentation less readable.